From: Ricardo Neri <ricardo.neri@ti.com>
Date: Sat, 15 Jan 2011 06:26:45 +0000 (-0600)
Subject: ASoC: ABE DAI: Mute/Unmute MIX_DL2_INPUT_MM_DL when MM_DL_LP is used
X-Git-Url: http://git.omapzoom.org/?p=kernel%2Fomap.git;a=commitdiff_plain;h=c96e270a05eadda558d3ce1b6529d85f95b3217e;hp=e9c656e4d51ade94dc286dd747a1b8b902d697e3

ASoC: ABE DAI: Mute/Unmute MIX_DL2_INPUT_MM_DL when MM_DL_LP is used

Low-power port is now only restricted to ping-pong buffer, not
to OPP25 anymore. Hence, it can be used with PDM_DL2 backend.

This patch is to mute/unmute MIX_DL1_INPUT_MM_DL as well as
MIX_DL2_INPUT_MM_DL when low-power port is used.

Change-Id: I41427a5304ed0a33abce28591150b5080a217a10
Signed-off-by: Ricardo Neri <ricardo.neri@ti.com>
---

Index: android-2.6.35/sound/soc/omap/omap-abe.c
===================================================================
--- android-2.6.35.orig/sound/soc/omap/omap-abe.c	2011-01-25 11:28:53.644092202 +0530
+++ android-2.6.35/sound/soc/omap/omap-abe.c	2011-01-25 11:49:44.857809984 +0530
@@ -1330,10 +1330,10 @@
 
 	switch(rtd->cpu_dai->id) {
 	case ABE_FRONTEND_DAI_MEDIA:
+	case ABE_FRONTEND_DAI_LP_MEDIA:
 		if (abe_data.be_active[OMAP_ABE_DAI_PDM_DL2][SNDRV_PCM_STREAM_PLAYBACK]) {
 			abe_mute_gain(MIXDL2, MIX_DL2_INPUT_MM_DL);
 		}
-	case ABE_FRONTEND_DAI_LP_MEDIA:
 		if (abe_data.be_active[OMAP_ABE_DAI_PDM_DL1][SNDRV_PCM_STREAM_PLAYBACK]) {
 			abe_mute_gain(MIXDL1, MIX_DL1_INPUT_MM_DL);
 		}
@@ -1372,10 +1372,10 @@
 
 	switch(rtd->cpu_dai->id) {
 	case ABE_FRONTEND_DAI_MEDIA:
+	case ABE_FRONTEND_DAI_LP_MEDIA:
 		if (abe_data.be_active[OMAP_ABE_DAI_PDM_DL2][SNDRV_PCM_STREAM_PLAYBACK]) {
 			abe_unmute_gain(MIXDL2, MIX_DL2_INPUT_MM_DL);
 		}
-	case ABE_FRONTEND_DAI_LP_MEDIA:
 		if (abe_data.be_active[OMAP_ABE_DAI_PDM_DL1][SNDRV_PCM_STREAM_PLAYBACK]) {
 			abe_unmute_gain(MIXDL1, MIX_DL1_INPUT_MM_DL);
 		}
